pkg/metrics/neutron.go

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: MIT
+
+package metrics
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/extensions/layer3/floatingips"
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/klog"
+)
+
+var (
+	neutronFloatingIPStatus    *prometheus.GaugeVec
+	neutronFloatingIPCreated   *prometheus.GaugeVec
+	neutronFloatingIPUpdatedAt *prometheus.GaugeVec
+
+	floatingIPLabels = []string{"id", "floating_ip", "fixed_ip", "port_id"}
+)
+
+func registerNeutronMetrics() {
+	neutronFloatingIPStatus = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: generateName("neutron_floating_ip_status"),
+			Help: "Neutron floating ip status",
+		},
+		floatingIPLabels,
+	)
+	neutronFloatingIPCreated = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: generateName("neutron_floatingip_created_at"),
+			Help: "Neutron floating ip created at",
+		},
+		floatingIPLabels,
+	)
+	neutronFloatingIPUpdatedAt = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: generateName("neutron_floatingip_updated_at"),
+			Help: "Neutron floating ip updated at",
+		},
+		floatingIPLabels,
+	)
+
+	prometheus.MustRegister(neutronFloatingIPStatus)
+	prometheus.MustRegister(neutronFloatingIPCreated)
+	prometheus.MustRegister(neutronFloatingIPUpdatedAt)
```

Comment:
We also need the floating IP quota.

Reply:
Also thought about it, will be an follow-up PR.